### PR TITLE
Add custom distance for smile detector demo

### PR DESCRIPTION
**Pull request by tolga-b**
This PR adds custom distance function for smile detector demo, such that the distance between two examples are computed with respect to their average color difference.
This is dependent on the https://github.com/tensorflow/tensorboard/pull/2607 for custom distance function support.

### CODE REVIEW

File: WIT_Smile_Detector.ipynb [hunk 1]
```diff
@@ -209,17 +212,33 @@
...
+        "  return im\n",
+        "\n",
+        "# Define the custom distance function that compares the average color of images\n",
+        "def image_mean_distance(ex, exs, params):\n",
+        "  selected_im = decode_image(ex)\n",
+        "  mean_color = np.mean(selected_im, axis=(0,1))\n",
+        "  image_distances = [np.linalg.norm(mean_color - np.mean(decode_image(e), axis=(0,1))) for e in exs]\n",
+        "  return image_distances\n",
+        "\n",
+        "config_builder = config_builder.set_custom_distance_fn(image_mean_distance)\n",
```
Comment (jameswex) [inline WIT_Smile_Detector.ipynb]: simpler code to define this above and then set it in the WitConfigBuilder line above

File: WIT_Smile_Detector.ipynb [hunk 2]
```diff
@@ -209,17 +212,33 @@
...
     },
     {
+      "cell_type": "markdown",
       "metadata": {
         "id": "A1s1_SiOyS0l",
         "colab_type": "text"
       },
-      "cell_type": "markdown",
       "source": [
         "#### Exploration ideas\n",
```
Comment (jameswex) [inline WIT_Smile_Detector.ipynb]: update exploration ideas given the new counterfactual goodness

File: WIT_Smile_Detector.ipynb (hunk 1 above)
Comment (tolga-b) [reply WIT_Smile_Detector.ipynb]: done.

File: WIT_Smile_Detector.ipynb (hunk 2 above)
Comment (tolga-b) [reply WIT_Smile_Detector.ipynb]: added two ideas, let me know if you have any extra thoughts.

**Comment by jameswex** [comment]
looks good but we'll need to wait until we pip release with the custom distance before pushing this.

**Pull request merged**